test/k8s-gen-bot-config.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+# Copyright 2023 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o xtrace
+
+# This script generates the config and rules required for testing the master branch of k/k
+# with publishing bot
+
+BOT_CONFIG_DIRECTORY="${1:-bot-configs}"
+
+mkdir "${BOT_CONFIG_DIRECTORY}"
+
+## generate the required config
+# use the content from configmap in the data section
+sed -e '1,/config: |/d' configs/kubernetes-configmap.yaml > "${BOT_CONFIG_DIRECTORY}"/config
+# remove leading white spaces from the generated file
+sed "s/^[ \t]*//" -i "${BOT_CONFIG_DIRECTORY}"/config
```

Comment:
This won't work on mac, see https://unix.stackexchange.com/questions/145299/simple-sed-replacement-of-tabs-mysteriously-failing